### PR TITLE
Update PHP settings on performance best practices software page

### LINKED ISSUE #8394: Update PHP settings on performance best practices software page
# General issue

## Description:

The recommendation provided in https://support.magento.com/hc/en-us/articles/360050422532 article contradicts with the 
recommendation given in https://devdocs.magento.com/guides/v2.4/performance-best-practices/software.html#php-settings. In Magento cloud environments  opcache.enable_cli setting has to be set to 0, without that the changes in .magento.env.yaml are not get updated into app/etc/env.php file.

## Possible solutions:

Can we add a note in this https://devdocs.magento.com/guides/v2.4/performance-best-practices/software.html#php-settings document to mentioned that this opcache.enable_cli setting has to be set 0 in Magento cloud environments.

### PR DESCRIPTION
**Pull request by chiranjeevi-cj**
## Purpose of this pull request

This pull request (PR) Updates PHP settings on the performance best practices software page

## Affected DevDocs pages

-  https://devdocs.magento.com/guides/v2.4/performance-best-practices/software.html#php-settings

Fixes #8394

### CODE REVIEW

File: src/guides/v2.4/performance-best-practices/software.md
Comment (BarnyShergold) [inline src/guides/v2.4/performance-best-practices/software.md:60]:
```suggestion
In Magento Cloud environments, `opcache.enable_cli` setting has to be set to `0`. Without this setting the changes in `.magento.env.yaml` will not get updated into the `app/etc/env.php` file.
```

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
Can the second reviewer double-check the technical truthfulness of this addition? I'm not aware that this setting has anything to do with updating env.php

**Review by BarnyShergold — COMMENTED** [review]
@hguthrie - Could do with a second opinion on the technical accuracy of this change. 'm not aware of a PHP setting having this effect

**Comment by hguthrie** [comment]
It may seem that adding this note will help, but after some research, I found that the next release of Cloud tools addresses this with a less manual approach. See #8240 
@bdenham For this issue, can you review whether the release note is sufficient or if anything further added to the Cloud guide, or other guide?

**Review by hguthrie — CHANGES_REQUESTED** [review]
This may not be necessary. Waiting for feedback related to release notes.

**Comment by bdenham** [comment]
Sorry I'm late to the party. Looking at this now and finding a second opinion about the accuracy. It is more useful than the release note because of the details it provides.

**Comment by bdenham** [comment]
I talked to @meker12 about this and as she said: " For Magento on-premise, the default for `opcache` is `1` and the core documentation is correct. This note might need to go in the Cloud documentation, because the people that are going to run into the problem are Cloud users."

Margaret also mentioned a support article for the issue: https://support.magento.com/hc/en-us/articles/360050422532

I'm still waiting to hear back from a Cloud developer, but at a minimum, it looks like this needs to be added to the Cloud Guide, not Performance Best Practices.

**Comment by meker12** [comment]
> I talked to @meker12 about this and as she said: " For Magento on-premise, the default for `opcache` is `1` and the core documentation is correct. This note might need to go in the Cloud documentation, because the people that are going to run into the problem are Cloud users."
> 
> Margaret also mentioned a support article for the issue: https://support.magento.com/hc/en-us/articles/360050422532
> 
> I'm still waiting to hear back from a Cloud developer, but at a minimum, it looks like this needs to be added to the Cloud Guide, not Performance Best Practices.

@hguthrie mentioned the 2.1.4 ECE tools improvement that sets the `opcache_cli` to the correct setting for Cloud by default (see #8240) . That's what I thought -- Cloud sets this to the right setting by default, so if people run into issues, then the support article is there. (Assumes that Cloud project uses ECE-tools 2.1.4 or later.)

**Comment by dobooth** [comment]
@bdenham @hguthrie any updates here? Ready to go?

**Comment by hguthrie** [comment]
@dobooth This has a support article that provides this information in further detail. The next release of Cloud tools contains corrections or additional information for this issue. If we add this note now, it has to be cleaned or updated for the Cloud release. @bdenham stated he is waiting for Cloud team feedback, so I will mark this PR as waiting. If @bdenham decides to proceed with this PR, then this change must be reviewed again during the Cloud docs release so that it agrees with the changes made by the Cloud team.

**Comment by bdenham** [comment]
Yeah, I'm going to close this for all the reasons mentioned above. The setting is now set correctly by default, so it would create more confusion if this was added. Regardless, thanks @chiranjeevi-cj for your heads-up work on this.

**Pull request closed without merging**
